lib/ui/form/widget/form_survey_answer_nps.dart

```diff
@@ -0,0 +1,124 @@
+    return FormSurveyAnswerNpsState();
+  }
+}
+
+class FormSurveyAnswerNpsState extends ConsumerState<FormSurveyAnswerNps> {
+  final selectedScoreProvider = StateProvider.autoDispose<int>((_) => 0);
+  late int scoresLength;
+
+  @override
+  void initState() {
+    super.initState();
+    scoresLength = widget.question.answers.length;
+    ref.read(selectedScoreProvider.notifier).state = (scoresLength / 2).floor();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final scores = List.generate(scoresLength, (index) => index);
+    final scoreState = ref.read(selectedScoreProvider.notifier);
+    final selectedScore = ref.watch(selectedScoreProvider);
+    final selectedStyle = Theme.of(context).textTheme.labelSmall;
+    final unselectedStyle = selectedStyle?.copyWith(color: AppColors.white50);
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        SingleChildScrollView(
+          scrollDirection: Axis.horizontal,
+          child: SizedBox(
+            height: AppDimensions.answerNpsHeight,
+            child: Container(
+              decoration: BoxDecoration(
+                border: Border.all(
+                  color: Colors.white,
+                  width: AppDimensions.answerNpsBorderWidth,
+                ),
+                borderRadius: const BorderRadius.all(
+                  Radius.circular(AppDimensions.radius10),
+                ),
+              ),
+              child: Row(
+                children: List.generate(scoresLength, (index) {
+                  final score = scores[index];
+                  return GestureDetector(
+                    onTap: () => scoreState.state = score,
+                    child: _buildScore(context, ref, score, scoresLength),
+                  );
+                }),
+              ),
+            ),
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(top: AppDimensions.spacing18),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text(
+                AppLocalizations.of(context)!.not_at_all_likely,
+                style: selectedScore <= scoresLength / 2
+                    ? selectedStyle
+                    : unselectedStyle,
+              ),
+              Text(
+                AppLocalizations.of(context)!.extremely_likely,
+                style: selectedScore > scoresLength / 2
+                    ? selectedStyle
+                    : unselectedStyle,
+              )
+            ],
+          ),
+        )
+      ],
+    );
+  }
+
+  Widget _buildScore(
+    BuildContext context,
+    WidgetRef ref,
+    int score,
+    int scoresLength,
+  ) {
+    final selectedScore = ref.watch(selectedScoreProvider);
+    final selectedStyle = Theme.of(context).textTheme.labelMedium;
+    final unselectedStyle = selectedStyle?.copyWith(color: AppColors.white50);
+
+    dynamic border;
```

Comment:
Can we declare it like this? 🤔 
```suggestion
    Border? border;
```

Reply:
@chornerman Updated 🚀